master/master.cfg

```diff
@@ -349,479 +350,311 @@ def prioritizeBuilders(buildmaster, builders):
-                "Ubuntu-14.04-x86_64-buildslave1", "BUILD",
-                bb_master, "ami-550ce335", bb_url,
-                ec2_default_access, ec2_default_secret, 
-                ec2_default_keypair_name),
-            EC2LargeSlaveInfo(
-                "Ubuntu-14.04-x86_64-buildslave2", "BUILD",
-                bb_master, "ami-550ce335", bb_url,
-                ec2_default_access, ec2_default_secret, 
-                ec2_default_keypair_name),
-        ],
+    ZFSBuilderConfig(
+        name="Ubuntu 14.04 x86_64 (BUILD)",
+        factory=build_factory,
+        slavenames=[slave.name for slave in ubuntu14_x86_64_slave],
         tags=build_distro_tags,
         properties=builder_default_properties,
     ),
-    BuilderInfo(
-        "Ubuntu 16.04 x86_64 (BUILD)", build_factory,
-        [
-            EC2LargeSlaveInfo(
-                "Ubuntu-16.04-x86_64-buildslave1", "BUILD",
-                bb_master, "ami-9f29c0ff", bb_url,
-                ec2_default_access, ec2_default_secret, 
-                ec2_default_keypair_name),
-            EC2LargeSlaveInfo(
-                "Ubuntu-16.04-x86_64-buildslave2", "BUILD",
-                bb_master, "ami-9f29c0ff", bb_url,
-                ec2_default_access, ec2_default_secret, 
-                ec2_default_keypair_name),
-        ],
+    ZFSBuilderConfig(
+        name="Ubuntu 16.04 x86_64 (BUILD)",
+        factory=build_factory,
+        slavenames=[slave.name for slave in ubuntu16_x86_64_slave],
         tags=build_distro_tags,
         properties=builder_default_properties,
     ),
     #### Architecture builders
-    BuilderInfo(
-        "Ubuntu 14.04 i686 (BUILD)", build_factory,
-        [
-            EC2PVLargeSlaveInfo(
-                "Ubuntu-14.04-i686-buildslave1", "BUILD",
-                bb_master, "ami-7d04ed1d", bb_url,
-                ec2_default_access, ec2_default_secret, 
-                ec2_default_keypair_name),
-            EC2PVLargeSlaveInfo(
-                "Ubuntu-14.04-i686-buildslave2", "BUILD",
-                bb_master, "ami-7d04ed1d", bb_url,
-                ec2_default_access, ec2_default_secret, 
-                ec2_default_keypair_name),
-        ],
+    ZFSBuilderConfig(
+        name="Ubuntu 14.04 i686 (BUILD)",
+        factory=build_factory,
+        slavenames=[slave.name for slave in ubuntu14_i686_slave],
         tags=build_arch_tags,
         properties=builder_default_properties,
     ),
-    BuilderInfo(
-        "Debian 8 arm (BUILD)", build_factory,
-        [
-            PersistSlaveInfo(
-                "Debian-8-arm-buildslave1",
-                slave_userpass["Debian-8-arm-buildslave1"]),
-            PersistSlaveInfo(
-                "Debian-8-arm-buildslave2",
-                slave_userpass["Debian-8-arm-buildslave2"]),
-        ],
+    ZFSBuilderConfig(
+        name="Debian 8 arm (BUILD)",
+        factory=build_factory,
+        slavenames=[slave.name for slave in debian8_arm_slave],
```

Comment:
This should be `slave.slavename`, I've made the fix locally and can squash it in to this patch.  No need to refresh this.  This was the only issues I've found thus far.